lib/src/commit.rs

```diff
@@ -163,6 +163,19 @@ impl Commit {
     }
 }
 
+pub trait CommitIteratorExt<'c, I> {
+    fn ids(self) -> Box<dyn Iterator<Item = &'c CommitId> + 'c>;
```

Comment:
nit: `impl Iterator<Item = &'c CommitId>` now works :)

Reply:
This will monomorphize and make the binary larger (by some small amount), right? I figured we don't really need the performance of monomorphized code here and a Box would be fine.

Should we avoid `Box` in general as a matter of style?

Reply:
Do we care for code size bloat when writing `.iter().map(something)`? I agree that the performance doesn't matter, but I feel boxing here is silly because it's just a short for `.map(|commit| commit.id())`.

Reply:
:D fair enough, we probably don't care. I removed the Box.